### PR TITLE
ci(cypress): add test cases for duplicate requests

### PR DESCRIPTION
**Pull request by likhinbopanna**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [x] CI/CD

## Description

 Added test cases for adding duplicaten ID's like `payment_id` , `refund_id` and `customer_id` in the requests 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Enhancement of test cases 

## How did you test it?

Variation test cases 
image

image

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/support/commands.js [hunk 1]
```diff
@@ -888,26 +888,38 @@ Cypress.Commands.add(
...
+          );
+          expect(
+            customerCreateBody.phone_country_code,
+            "phone_country_code"
+          ).to.equal(response.body.phone_country_code);
+        });
+      } else if (response.status === 400) {
+        expect(response.body.error.code).to.equal("IR_12");
+        expect(response.body.error.message).to.equal(
+          "Customer with the given `customer_id` already exists"
```
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:920]: there exist many other 4xx errors. this assertion will always be checked if status is 400. you need to validate before checking these assertions.

first verify if the error message is related to duplication. you can do something like:

```js
if (response.body.error.message).includes("already exists") {
	// check if it is the `customer_id` that exist and then make the assertions accordingly
	// also, remember to add names to that. `(...error.code, "error_code")`
}
```

File: cypress-tests/cypress/support/commands.js [hunk 2]
```diff
@@ -888,26 +888,38 @@ Cypress.Commands.add(
         "api-key": globalState.get("apiKey"),
       },
       body: customerCreateBody,
+      failOnStatusCode: false,
     }).then((response) => {
-      globalState.set("customerId", response.body.customer_id);
-      logRequestId(response.headers["x-request-id"]);
+      if (response.status === 200) {
+        globalState.set("customerId", response.body.customer_id);
+        logRequestId(response.headers["x-request-id"]);
```
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:895]:
```suggestion
        logRequestId(response.headers["x-request-id"]);
        globalState.set("customerId", response.body.customer_id);
```

File: cypress-tests/cypress/e2e/spec/Payment/00022-Variations.cy.js [hunk 3]
```diff
@@ -702,4 +702,117 @@ describe("Corner cases", () => {
...
+
+    it("Create new customer", () => {
+      cy.createCustomerCallTest(fixtures.customerCreateBody, globalState);
+    });
+
+    it("Create a customer with a duplicate customer ID", () => {
+      const cusdata = fixtures.customerCreateBody;
+      cusdata.customer_id = globalState.get("customerId");
+
+      cy.createCustomerCallTest(cusdata, globalState);
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/spec/Payment/00022-Variations.cy.js:815]:
```suggestion
      const customerData = fixtures.customerCreateBody;
      customerData.customer_id = globalState.get("customerId");

      cy.createCustomerCallTest(customerData, globalState);
```

File: cypress-tests/cypress/support/commands.js (hunk 1 above)
Comment (pixincreate) [reply cypress-tests/cypress/support/commands.js:920]: also, add an `else` condition at the end that actually throws an error if we get anything other than the mentioned we'll get to know about it

**Review by Gnanasundari24 — APPROVED** [review]

**Review by pixincreate — APPROVED** [review]
